### PR TITLE
Migrate to `hatch`

### PR DESCRIPTION
**Pull request by jtpio**
Aligning the packaging story with the other projects: https://blog.jupyter.org/packaging-for-jupyter-in-2022-c7be64c38926 

Although `empack` is not strictly related to Jupyter it's nice to streamline the setup to avoid fatigue over time.

One can use the `hatch version` command to bump version:

image

### CODE REVIEW

**Comment by jtpio** [comment]
cc @martinRenou for review and for triggering the CI run

**Comment by jtpio** [comment]
ping @martinRenou

I checked on my fork and CI seems to be passing: https://github.com/jtpio/empack/pull/1

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+]
+dependencies = [
+    "appdirs",
+    "networkx",
+    "pydantic",
+    "pydantic",
```
Comment (martinRenou) [inline pyproject.toml:31]: pydantic is there twice, it was already the case before but since we're touching this file let's fix it

```suggestion
```

**Review by martinRenou — CHANGES_REQUESTED** [review]
Thanks!

**Review by martinRenou — APPROVED** [review]
LGTM! Thanks

**Pull request merged**